pkg/ipam/ipam_linux.go

```diff
@@ -52,6 +58,21 @@ func ConfigureIface(ifName string, res *current.Result) error {
 			return fmt.Errorf("failed to add IP addr %v to %q: invalid interface index", ipc, ifName)
 		}
 
+		// Make sure sysctl "disable_ipv6" is 0 if we are about to add
+		// an IPv6 address to the interface
+		if !has_enabled_ipv6 && ipc.Version == "6" {
+			// Enable IPv6 for the interface being configured as well as
+			// loopback "lo", "all" and "default"
```

Comment:
Any particular reason to do this for lo, all, and default?  It may not be wanted/needed on other interfaces created by other CNI chains executing for the same container.

Reply:
You are right, for `all` and `default` I do not have any particularly strong reason. My reasoning was that it would be better to have the default to be "IPv6-enabled" but this is not something CNI need to enforce.

Regarding the loopback interface `lo` I could not get the kube-dns pod to run without IPv6 enabled on `lo`. If I understood correctly, kube-dns uses 3 containers sharing the same network NS, and the containers use the loopback device for internal communication. When running kube-dns with IPv6 it expects `::1` (IPv6 on loopback) to be available.

I think it is fairly reasonable to assume that applications running IPv6 would expect IPv6 to be available on loopback.

I have updated the pull-request to only enable IPv6 for `lo` and the interface being configured by CNI. Thanks for the feedback!